src/harness/testcases/WINNF_FT_S_MCP_testcase.py

```diff
@@ -0,0 +1,378 @@
+        device_2 = json.load(
+          open(os.path.join('testcases', 'testdata', 'device_b.json')))
+        device_3 = json.load(
+          open(os.path.join('testcases', 'testdata', 'device_c.json')))
+        device_4 = json.load(
+          open(os.path.join('testcases', 'testdata', 'device_d.json')))
+        device_5 = json.load(
+          open(os.path.join('testcases', 'testdata', 'device_e.json')))
+  
+        # Load Grant requests
+        grant_request_1 = json.load(
+          open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+        grant_request_2 = json.load(
+          open(os.path.join('testcases', 'testdata', 'grant_1.json')))
+        grant_request_3 = json.load(
+          open(os.path.join('testcases', 'testdata', 'grant_2.json')))
+        grant_request_4 = json.load(
+          open(os.path.join('testcases', 'testdata', 'grant_3.json')))
+      
+        # Load GWPZ Record
+        gwpz_record_1 = json.load(
+          open(os.path.join('testcases', 'testdata', 'gwpz_record_0.json')))
+  
+        # Load FSS record
+        fss_record_1 = json.load(
+          open(os.path.join('testcases', 'testdata', 'fss_record_0.json')))
+  
+        # Load ESC record
+        esc_record_1 = json.load(
+          open(os.path.join('testcases', 'testdata', 'esc_sensor_record_0.json')))
+     
+        # Load PPA record
+        ppa_record_1 = json.load(
+          open(os.path.join('testcases', 'testdata', 'ppa_record_0.json'))) 
+        # Registration configuration for iteration 1
+        cbsd_records_iter_1 = {'registrationRequests': [device_1, device_2],
+                              'grantRequests': [grant_request_1, grant_request_2]}
+        cbsd_records_iter_2 = {'registrationRequests': [device_3, device_4],
+                              'grantRequests': [grant_request_3, grant_request_4]}
+        protected_entities_iter_1 = {'PPA': [ppa_record_1], 'ESC': [esc_record_1]}
+        protected_entities_iter_2 = {'GWPZ': [gwpz_record_1] , 'FSS': [fss_record_1]}
+
+
+        # SAS Test Harness FAD records
+        sas_th_cbsd_0 = json.load(
+          open(os.path.join('testcases', 'testdata', 'cbsd_0.json')))
+        sas_th_cbsd_1 = json.load(
+          open(os.path.join('testcases', 'testdata', 'cbsd_1.json')))
+        sas_th_cbsd_2 = json.load(
+          open(os.path.join('testcases', 'testdata', 'cbsd_2.json')))
+        sas_th_cbsd_3 = json.load(
+          open(os.path.join('testcases', 'testdata', 'cbsd_3.json')))
+        sas_th_cbsd_4 = json.load(
+          open(os.path.join('testcases', 'testdata', 'cbsd_4.json')))
+        sas_th_cbsd_5 = json.load(
+          open(os.path.join('testcases', 'testdata', 'cbsd_5.json')))
+
+        sas_th_0_cbsd_iter_0 = [sas_th_cbsd_0]
+        sas_th_0_cbsd_iter_1 = [sas_th_cbsd_1]
+        sas_th_0_cbsd_iter_2 = [sas_th_cbsd_2]
+        sas_th_1_cbsd_iter_0 = [sas_th_cbsd_3]
+        sas_th_1_cbsd_iter_1 = [sas_th_cbsd_4]
```

Comment:
Building up the data inside the SAS TH, instead of constantly swapping it out, would probably be a better test of the system. It would be ideal if the default config looked more like what we believe to be a reasonable test case.

Reply:
This   configuration is for SAS TH. Parameters are configured based on the inputs   received from the SAS TH development. Which is is under #PR368, we will   update the SAS TH configurations section after #PR386 merge

Reply:
Good point. I will review the SAS TH configuration in #368 instead, and focus the comments for this PR on the issues which are specific to MCP. I suspect we will submit that PR first anyway so it should not cause any problems.